packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -452,13 +481,66 @@ class AndroidCameraCameraX extends CameraPlatform {
 
   /// Gets the supported step size for exposure offset for the selected camera in EV units.
   ///
-  /// Returns 0 when exposure compensation is not supported.
-  ///
   /// [cameraId] not used.
   @override
   Future<double> getExposureOffsetStepSize(int cameraId) async {
     final ExposureState exposureState = await cameraInfo!.getExposureState();
-    return exposureState.exposureCompensationStep;
+    final double exposureOffsetStepSize =
+        exposureState.exposureCompensationStep;
+    if (exposureOffsetStepSize == 0) {
+      // CameraX returns a step size of 0 if exposure compensation is not
+      // supported for the device.
+      throw CameraException(exposureCompensationNotSupported,
+          'Exposure compensation not supported for the device.');
+    }
+    return exposureOffsetStepSize;
+  }
+
+  /// Sets the exposure offset for the selected camera.
+  ///
+  /// The supplied [offset] value should be in EV units. 1 EV unit represents a
+  /// doubling in brightness. It should be between the minimum and maximum offsets
+  /// obtained through `getMinExposureOffset` and `getMaxExposureOffset` respectively.
+  /// Throws a `CameraException` when an illegal offset is supplied.
+  ///
+  /// When the supplied [offset] value does not align with the step size obtained
+  /// through `getExposureStepSize`, it will automatically be rounded to the nearest step.
+  ///
+  /// Returns the (rounded) offset value that was set.
+  @override
+  Future<double> setExposureOffset(int cameraId, double offset) async {
+    final double exposureOffsetStepSize =
+        (await cameraInfo!.getExposureState()).exposureCompensationStep;
+    if (exposureOffsetStepSize == 0) {
+      throw CameraException(exposureCompensationNotSupported,
+          'Exposure compensation not supported');
+    }
+
+    final int roundedExposureCompensationIndex =
+        (offset / exposureOffsetStepSize).round();
+
+    try {
+      await cameraControl!
+          .setExposureCompensationIndex(roundedExposureCompensationIndex);
+    } on PlatformException catch (e) {
+      throw CameraException(
+          'setExposureOffsetFailed',
+          e.message ??
+              'Setting the camera exposure compensation index failed.');
+    }
```

Comment:
Looking for feedback on throwing an exception here and forwarding the exception from `CameraControl` in order to conform with the platform interface, in the sense that the method

> Throws a `CameraException` when an illegal offset is supplied.

as per its documentation. Another option is explicitly checking in this method, but was worried that such an asynchronous call could impact performance.

Reply:
I'm fine with throwing an exception here since a user is explicitly warned about it in the app-facing package.

Although this does highlight the need for a way to check if a feature is supported without using a try/catch for every method. Something we will consider when the camera platform interface is updated.